packages/block-library/src/gallery/index.php

```diff
@@ -50,9 +50,13 @@ function block_core_gallery_render( $attributes, $content ) {
 	// because we only want to match against the value, not the CSS attribute.
 	if ( is_array( $gap ) ) {
 		foreach ( $gap as $key => $value ) {
+			// Make sure $gap[ $key ] is a string to avoid PHP 8.1 deprecation error in preg_match() when the value is null.
+			$gap[ $key ] = is_string( $gap[ $key ] ) ? $gap[ $key ] : '';
```

Comment:
Should this ternary use `$value` instead of `$gap[ $key ]`?

Otherwise, as far as I can tell, if `$value` is null, it is still passed to `preg_match()` unchanged.

Reply:
You are correct! Pushed a fix for that, thanks for spotting it 👍